presto-native-execution/presto_cpp/main/PrestoServer.h

```diff
@@ -214,7 +221,7 @@ class PrestoServer {
   std::unique_ptr<TaskManager> taskManager_;
   std::unique_ptr<TaskResource> taskResource_;
   std::atomic<NodeState> nodeState_{NodeState::kActive};
-  std::atomic_bool shuttingDown_{false};
+  folly::Synchronized<bool> shuttingDown_{false};
```

Comment:
Why is this change?

Reply:
@mbasmanova 

It is mentioned in the PR's description: "Code changed a bit to ensure we change the shuttingDown flag and the state safely."

The reason is that we don't want to attach the worker back if we are truly shutting down, so the two member variables need to be changed under a single lock.